### DISCUSSION TITLE
Unable to access web interface

### DISCUSSION THREAD

**Discussion by jjermany** [marcone/teslausb#897] [Q&A]
It works when the pi is connected to the home network but when attempting to connect on the road it does not. I connect to the oi wifi and attempt to access the IP. Would be nice to be able to grab important files right then and there if something were to happen.

**Comment by marcone** [comment]
How are you trying to access the Pi's web interface after connecting to its AP? I found that on my Android phone I need to browse to it by IP address (which would be 192.168.66.1 if you didn't change the default), because Chrome won't connect to it by name.

**Reply by jjermany** [reply]
That's what I try to do. Currently trying to reconfigure the API to keep the car awake. But after I'll test again.

**Reply by marcone** [reply]
> That's what I try to do. Currently trying to reconfigure the API to keep the car awake. But after I'll test again.

I'd love to be proven wrong, but I don't think the Tesla API works for TeslaUSB after Tesla's recent changes.

**Reply by jjermany** [reply]
I think you are right. I tried using sentry instead of stream and it's not working. I did confirm the key is working as it does pull all of my cars information. I'll have to add some extra lines to get it to work with TeslaFi correct?

**Reply by jjermany** [reply]
And lastly, Ive been formatting and reinstalling every time. How do I run the changes instead of fresh install? I need to save changed and just drag the file into the boot?

**Reply by jjermany** [reply]
Awesome. I think I have everything working! Thank you for your help!

**Reply by marcone** [reply]
> And lastly, Ive been formatting and reinstalling every time. How do I run the changes instead of fresh install? I need to save changed and just drag the file into the boot?

https://github.com/marcone/teslausb/wiki/FAQ#q-how-do-i-change-the-teslausb-configuration

**Reply by jjermany** [reply]
> > That's what I try to do. Currently trying to reconfigure the API to keep the car awake. But after I'll test again.
> 
> I'd love to be proven wrong, but I don't think the Tesla API works for TeslaUSB after Tesla's recent changes.

How can we change the wake up command to turn sentry on/off when using teslafi? It currently uses wake and sleep and I do not think that is enough to keep the USB's powered on.

**Reply by marcone** [reply]
From TeslaFi's list of commands at [about.teslafi.com/alexa ](https://about.teslafi.com/alexa/) it doesn't seem like turning on/off Sentry is something that TeslaFi supports. It appears that Tessie does though.
Honestly I think with the current state of things, the best solution is to either hardware the Pi to power, or use a passthrough battery bank. I'll be more functional and cheaper than a TeslaFi/Tessie subscription.

**Reply by jjermany** [reply]
![Screenshot_20241111-110804.png](https://github.com/user-attachments/assets/58ba9bf6-0de6-46fb-85a1-b95857faa1ce)

TeslaFi commands page when using the API Token.

**Reply by marcone** [reply]
My bad for trusting Google's AI summary, which sent me to that list of Alexa commands as the canonical list of commands that TeslaFi supports.
In any case, it looks like it wouldn't be too hard to add support for that. I have no way of verifying it though since I'm not a TeslaFi subscriber, so you'd have to do it yourself.
There's already code in TeslaUSB's `awake_start` and `awake_stop` scripts to use TeslaFi's "wake_up" command, so you can look at that to see how to call the TeslaFI API, but use the sentry command URL instead.
You can look at the existing code that used the Tesla API directly to turn on/off Sentry mode to see where to add TeslaFi support.

**Reply by jjermany** [reply]
Okay, so I have updated both awake_start and awake_stop files and added an export TESLAFI_AWAKE_MODE=sentry. I'm not sure exactly how to install it using the instructions in the FAQ. I am assuming I can edit the config file, drop it in the root, and run a shell script to rerun the install.

**Reply by marcone** [reply]
If you've updated the awake_* scripts and the config file on the Pi, then just rebooting it should be enough to apply the changes.

**Reply by jjermany** [reply]
Okay awesome. I will try when the car gets back home and see if it works. Thanks a lot!

**Reply by jjermany** [reply]
I cannot find these files on the pi. I may need to do a fresh install.

**Reply by jjermany** [reply]
I made those changes on my computer. Not seeing them on the actual pi when I plug it in to the computer.

**Reply by marcone** [reply]
What do you mean by that? If you plug the Pi into your computer you will see the "CAM" drive that the car sees when you plug the Pi into the car, i.e. the drive that TeslaUSB emulates and which the car records dashcam videos to.

You need to ssh into the Pi to see the scripts that TeslaUSB uses, in /root/bin.

**Reply by jjermany** [reply]
Thank you for that. I was able to edit the files using nano. It now keeps the car awake with Sentry mode and turns off after archiving.

**Reply by marcone** [reply]
Would you like to contribute those changes back to the project?

**Reply by jjermany** [reply]
Sure I can do that in the next couple of days. How do I add the "teslausb" host name for the access point in the car? It seems that hostname only works when connected to our home network. Maybe it's a silly question but this is all kinda new to me.

**Reply by marcone** [reply]
> How do I add the "teslausb" host name for the access point in the car?

I'm not sure. I noticed that from an Android phone I can only access the Pi via its IP address when connected to its AP, but with an iPhone it works when accessed via hostname. Could be the Pi is already doing the right thing and Android/Chrome is just being dumb.

**Reply by marcone** [reply]
> Sure I can do that in the next couple of days. 

Would be nice if you could contribute that back to the project, since Tesla/TeslaFi has now limited the number of "wake" commands that can be issued per month, and will start charging for overages.

**Reply by miles267** [reply]
@jjermany would you please merge this functionality back to the project? Thank you.

**Reply by jjermany** [reply]
> @jjermany would you please merge this functionality back to the project? Thank you.

I just sent up the change. I am pretty new to all the github functionality so Ill do my best here. 

Thanks,

**Reply by marcone** [reply]
> > @jjermany would you please merge this functionality back to the project? Thank you.
> 
> I just sent up the change. I am pretty new to all the github functionality so Ill do my best here.
> 
> Thanks,

Looks like you didn't actually create a pull request, and the [change you made in your fork](https://github.com/jjermany/teslausb-1/commit/f45ec25cfd249bfb0b0ef3b2a63e6f85d98afb04) only modifies awake_start to turn sentry mode on, but doesn't change awake_stop to turn it back off afterward.
(the change also includes a bunch of unrelated whitespace changes, log text changes, and what looks like debug code)
